### ISSUE TITLE
Expected 'AS' in alias when SELECT

### ISSUE THREAD

**Issue by Catwoolfii** [timeplus-io/proton#611]
I'm just starting to try Proton and maybe I'm missing something or misunderstanding something.
I'm trying to run a request with pseudocode and getting an exception. This behavior is not reproduced in any version of Clickhouse.

**How to reproduce**

1. Executing a query:
`SELECT x.* FROM (SELECT 'test' name) x`

When executing query following exception is received:
```
Syntax error: failed at position 32 ('name'):

SELECT x.* FROM (SELECT 'test' name) x

Expected one of: NOT, BETWEEN, IS, AS, FROM, WHERE, PARTITION BY, SHUFFLE BY, GROUP BY, WITH, HAVING, WINDOW, ORDER BY, LIMIT, EMIT, SETTINGS, UNION
```
2. Executing a query when an alias is created with the 'AS' keyword:
`SELECT x.* FROM (SELECT 'test' AS name) AS x`

When executing query following output is received:
```
┌─name─┐
│ test │
└──────┘

1 row in set. Elapsed: 0.001 sec.
```

**Additional context**
I expect that such queries should be executed without exceptions, since the keyword 'AS' is optional.

**Comment by Sake169** [comment]
yes, it may need an enhancement, i tried the clickhouse in the same query, it can work well,

**Comment by yl-lisen** [comment]
Thank you for your experience @Catwoolfii . 
For some reasons, we must use explicit aliases for tables via `AS` keyword.
Usually, we always recommend using `AS` to define aliases everywhere, which is more understandable.

**Comment by yokofly** [comment]
fixed in latest proton 3.0 version
```
~ docker exec -it gen proton-client   
proton client version 3.0.3.
Connecting to localhost:8463 as user default.
Connected to proton server version 3.0.3 revision 54473.

proton :) SELECT x.* FROM (SELECT 'test' name) x
          

SELECT
  x.*
FROM
  (
    SELECT
      'test' AS name
  ) AS x

Query id: a05db00b-a780-4831-a6fb-0228ad0d9a92

┌─name─┐
│ test │
└──────┘

1 row in set. Elapsed: 0.001 sec. 

proton :) 


```
